### PR TITLE
[AEA-1005] add all non-optional fields in aea-config

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR makes the non-framework and non-optional config explicit in aea-config.yaml

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -29,28 +29,33 @@ skills:                                         # The list of skill public ids t
...
 max_reactions: 20                               # The maximum number of envelopes processed per call to `react` (only relevant for the `sync` mode)
 skill_exception_policy: propagate               # The exception policy applied to skills (must be one of "propagate", "just_log", or "stop_and_exit")
 connection_exception_policy: propagate          # The exception policy applied to connections (must be one of "propagate", "just_log", or "stop_and_exit")
-default_routing: {}                             # The default routing scheme applied to envelopes sent by the AEA, it maps from protocol public ids to connection public ids (both keys and values must satisfy PUBLIC_ID_REGEX)
 loop_mode: async                                # The agent loop mode (must be one of "sync" or "async")
 runtime_mode: threaded                          # The runtime mode (must be one of "threaded" or "async") and determines how agent loop and multiplexer are run
+decision_maker_handler: None                    # The decision maker handler to be used.
 ```
 
+The `aea-config.yaml` can further be extended with component configuration overrides:
```
Comment (DavidMinarsch) [inline docs/config.md:56]: @marcofavorito for you to fill in as part of your PR today

File: aea/configurations/base.py [hunk 2]
```diff
@@ -1224,9 +1224,14 @@ def __init__(
     @property
     def package_dependencies(self) -> Set[ComponentId]:
         """Get the connection dependencies."""
-        return set(
+        return {
             ComponentId(ComponentType.PROTOCOL, protocol_id)
             for protocol_id in self.protocols
+        }.union(
```
Comment (DavidMinarsch) [inline aea/configurations/base.py:1230]: missing previously, now fixed

File: aea/configurations/base.py [hunk 3]
```diff
@@ -1845,18 +1850,18 @@ def json(self) -> Dict:
...
+                },
+                "connection_private_key_paths": self.connection_private_key_paths_dict,
                 "private_key_paths": self.private_key_paths_dict,
+                "logging_config": self.logging_config,
                 "registry_path": self.registry_path,
                 "component_configurations": self.component_configurations_json(),
             }
         )  # type: Dict[str, Any]
 
-        if len(self.connection_private_key_paths_dict) > 0:
```
Comment (DavidMinarsch) [inline aea/configurations/base.py:1855]: both default_routing and connection_private_key_paths are important configurations of the agent behaviour. So they need to be present.

File: packages/fetchai/agents/aries_alice/aea-config.yaml [hunk 4]
```diff
@@ -31,3 +31,4 @@ registry_path: ../packages
 default_routing:
   fetchai/http:0.7.0: fetchai/http_client:0.10.0
   fetchai/oef_search:0.8.0: fetchai/soef:0.10.0
+connection_private_key_paths: {}
```
Comment (marcofavorito) [inline packages/fetchai/agents/aries_alice/aea-config.yaml:34]: this could be put just below `private_key_paths`

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**